### PR TITLE
Add more strictness to datatypes.

### PR DESCRIPTION
**Pull request by qnikst**

### CODE REVIEW

**Comment by facundominguez** [comment]
It looks like using force [here](https://github.com/tweag/distributed-process/pull/10/files) eliminates the need for this PR. Am I wrong?

**Comment by qnikst** [comment]
I yes, we don't need this commit in order to solve the issue. However this change may be in spirit of the `cloud-haskell` where we have our basic construct strict with `NFData` instances.

**Comment by qnikst** [comment]
I yes, we don't need this commit in order to solve the issue. However this change may be in spirit of the `cloud-haskell` where we have our basic construct strict with `NFData` instances.

**Comment by qnikst** [comment]
@facundominguez what to do with this PR, close as it's not required for the current use?

**Comment by qnikst** [comment]
@facundominguez please take another look, this patch could be used instead of forcing in https://github.com/haskell-distributed/distributed-process/pull/200

File: src/Control/Distributed/Static.hs [hunk 1]
```diff
@@ -293,7 +296,7 @@ getStaticLabel :: Get StaticLabel
 getStaticLabel = do
   header <- getWord8
   case header of
-    0 -> StaticLabel <$> get
+    0 -> mkStaticLabel <$> get
```
Comment (facundominguez) [inline src/Control/Distributed/Static.hs]: No need to force here.

File: src/Control/Distributed/Static.hs [hunk 2]
```diff
@@ -302,7 +305,7 @@ getStaticLabel = do
 -- It is the responsibility of the client code to make sure the corresponding
 -- entry in the 'RemoteTable' has the appropriate type.
 staticLabel :: String -> Static a
-staticLabel = Static . StaticLabel
+staticLabel = Static . mkStaticLabel
```
Comment (facundominguez) [inline src/Control/Distributed/Static.hs]: Inline `mkStaticLabel` and remove it?

**Comment by qnikst** [comment]
Fixed both

**Comment by facundominguez** [comment]
I don't see any changes.

**Comment by qnikst** [comment]
Is it better now?

**Comment by facundominguez** [comment]
Merged as 953fbde.

**Pull request closed without merging**